server/geoip_service.py

```diff
@@ -193,11 +193,16 @@ def load_db(self) -> None:
             # Set the time first, if the file is corrupted we don't need to try
             # loading it again anyways
             self.db_update_time = datetime.now()
-            self.db = geoip2.database.Reader(self.file_path)
+            new_db = maxminddb.open_database(self.file_path)
```

Comment:
If it were me I would wrap only this line in `try`/`except`, as this is the line that looks like it raises? Do we want to update `self.db_update_time` if opening the database might fail immediately after? I'd probably put the `open_database` call in `try`/`except` and then the rest in an `else`.

Reply:
In the grand scheme of things it doesn't matter if the `db_update_time` is updated before or after. If we update it after then we might generate more errors in the log tho if the database file is corrupted, as the service will keep trying to load it every 10 minutes.